File: c++/1.1 Herencia Simple y Modificadores de acceso protegidos/1.1.2 Modificadores/PersonaV3Test.cpp
#include <iostream>

using namespace std;

class PersonaV3 {
  /*                - MODIFICADORES DE ACCESO -
  +------------------------------------------------------------------+
  |   Modificador    |   Clase   | SubClase  |  Paquete  |    Todos  |
  +------------------+-----------+-----------+-----------+-----------|
  | public           |     sí    |    sí     |     sí    |    sí     |
  +------------------+-----------+-----------+-----------+-----------|
  | private          |     sí    |    no     |     no    |    no     |
  +------------------+-----------+-----------+-----------+-----------|
  | protected        |     sí    |    sí     |     sí    |    no     |
  +------------------+-----------+-----------+-----------+-----------|
  */

  private:
    string nombre;
    int edad;
    float estatura;

  public: 
    string getNombre() {
      return this->nombre;
    }

    void setNombre(string nombre) {
        this->nombre = nombre;
    }

    int getEdad() {
      return this->edad;
    }

    void setEdad(int edad) {
      this->edad = edad;
    }

    float getEstatura() {
      return this->estatura;
    }

    void setEstatura(float estatura) {
      this->estatura = estatura;
    }

    string toString() {
      return "{nombre='" + nombre + "'" + 
            ", edad='" + std::to_string(edad) + "'" + 
            ", estatura='" + std::to_string(estatura) + "'" + 
            "}";
    }
};

int main(int argc, char const *argv[])
{
  PersonaV3 persona = PersonaV3();
  persona.setNombre("");
  persona.setEdad(-4);
  persona.setEstatura(313.0f);

  cout << persona.toString() << endl;
  // {nombre='', edad='-4', estatura='313.0'}
  
  return 0;
}

File: python/1.1 Herencia Simple y Modificadores de acceso protegidos/1.1.2 Modificadores/PersonaV3Test.py
class PersonaV3:
  # atributos privados (su nombre inicia con dos guiones bajos)
  __nombre = ""
  __edad = 0
  __estatura = 0.0
  
  def getNombre(self): 
    return self.__nombre

  def setNombre(self, nombre): 
    self.__nombre = nombre

  def getEdad(self): 
    return self.__edad

  def setEdad(self, edad):
    self.__edad = edad

  def getEstatura(self): 
    return self.__estatura

  def setEstatura(self, estatura):
    self.__estatura = estatura
  
  def __str__(self):
    return f"[nombre='{self.__nombre}', edad='{self.__edad}', estatura='{self.__estatura}']"

# Uso de la clase Persona
if __name__ == "__main__":
  persona = PersonaV3()
  persona.setNombre("")
  persona.setEdad(-4)
  persona.setEstatura(313.0)
  
  print(persona)
  # [nombre='', edad='-4', estatura='313.0']